### DISCUSSION TITLE
A few basics

### DISCUSSION THREAD

**Discussion by rhaynes74** [firedrakeproject/firedrake#3513] [Firedrake support]
Hi folks, I have a few basic questions about obtaining solution traces and enforcing boundary conditions.
Suppose I have solved a 2d problem with the solution stored in function U.  

I would like to obtain a (Dirichlet) trace of the solution in the domain and use it as a Dirichlet boundary condition for another PDE solve.    What is the best way of doing this?   Can assume (if it makes a difference) that the PDEs are on compatible meshes (same number and orientation of elements).  

Related - suppose I want to provide the boundary data pointwise in an array, maybe loaded from a datafile , how is this best achieved. 

Sorry for the basic questions!

**Comment by colinjcotter** [comment]
On the same mesh, you can just provide U as the value to DirichletBC. On different meshes, you will need to interpolate from one mesh to the other.

From: rhaynes74 ***@***.***>
Date: Friday, 19 April 2024 at 15:12
To: firedrakeproject/firedrake ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [firedrakeproject/firedrake] A few basics (Discussion #3513)

Hi folks, I have a few basic questions about obtaining solution traces and enforcing boundary conditions.
Suppose I have solved a 2d problem with the solution stored in function U.

I would like to obtain a (Dirichlet) trace of the solution in the domain and use it as a Dirichlet boundary condition for another PDE solve. What is the best way of doing this? Can assume (if it makes a difference) that the PDEs are on compatible meshes (same number and orientation of elements).

Related - suppose I want to provide the boundary data pointwise in an array, maybe loaded from a datafile , how is this best achieved.

Sorry for the basic questions!

**Reply by rhaynes74** [reply]
I am not sure what is happening on the hood of interpolate, but is it necessary to interpolate onto the whole of the mesh for the second problem?   In some instances I could imagine that the mesh for the second problem may extend past the mesh for the first problem.

**Comment by colinjcotter** [comment]
No it isn’t. This is all described here:

https://www.firedrakeproject.org/interpolation.html#id5

From: rhaynes74 ***@***.***>
Date: Friday, 19 April 2024 at 15:38
To: firedrakeproject/firedrake ***@***.***>
Cc: Cotter, Colin J ***@***.***>, Comment ***@***.***>
Subject: Re: [firedrakeproject/firedrake] A few basics (Discussion #3513)

I am not sure what is happening on the hood of interpolate, but is it necessary to interpolate onto the whole of the mesh for the second problem? In some instances I could imagine that the mesh for the second problem may extend past the mesh for the first problem.
